### PR TITLE
Solution change to split out CLI and core command functions as separa…

### PR DESCRIPTION
**Pull request by guyacosta**
…te projects for improved NuGet support.

### CODE REVIEW

**Comment by guyacosta** [comment]
#87

File: RulesEngine/AppInspector.RulesEngine.csproj [hunk 1]
```diff
@@ -15,7 +15,7 @@
     <RepositoryUrl>https://github.com/Microsoft/ApplicationInspector</RepositoryUrl>
     <PackageReleaseNotes>1.0 General Release</PackageReleaseNotes>
```
[the remark this answers is not in the document]
Comment (gfs) [reply RulesEngine/AppInspector.RulesEngine.csproj]: Is this a package that will be distributed separately? Does it need these package fields at all?

File: AppInspector/AppInspector.Lib.csproj [hunk 2]
```diff
@@ -0,0 +1,82 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <PackageProjectUrl>https://github.com/microsoft/ApplicationInspector</PackageProjectUrl>
+    <RepositoryUrl>https://github.com/microsoft/ApplicationInspector</RepositoryUrl>
+    <RepositoryType></RepositoryType>
+    <PackageTags>Security Static Analyzer</PackageTags>
+    <PackageReleaseNotes>1.0 General Release</PackageReleaseNotes>
```
[the remark this answers is not in the document]
Comment (gfs) [reply AppInspector/AppInspector.Lib.csproj]: Either use a URL for release notes, populate this with something dynamically, or drop the field.

File: AppInspector/AppInspector.csproj [hunk 3]
```diff
@@ -1,126 +1,4 @@
...
-    <ProjectReference Include="..\RulesEngine\RulesEngine.csproj" />
-  </ItemGroup>
-
-
-  <ProjectExtensions><VisualStudio><UserProperties /></VisualStudio></ProjectExtensions>
-
-</Project>
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="Current" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup />
```
[the remark this answers is not in the document]
Comment (guyacosta) [reply AppInspector/AppInspector.csproj]: Old project name prior to this commit.  New split creates the .CLI and .Lib versions.  Not needed and good catch.

File: AppInspector/AppInspector.Lib.csproj [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    <Product>Application Inspector</Product>
+    <Authors>Microsoft</Authors>
+    <Version>1.0.28</Version>
+    <AssemblyName>AppInspectorLib</AssemblyName>
+    <RootNamespace>ApplicationInspector</RootNamespace>
+    <StartupObject></StartupObject>
+    <Company>Microsoft</Company>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyVersion>1.0.28.0</AssemblyVersion>
+    <FileVersion>1.0.28.0</FileVersion>
```
[the remark this answers is not in the document]
Comment (guyacosta) [reply AppInspector/AppInspector.Lib.csproj]: Removing it prevents the app from reporting the version when running i.e. by leaving it until the buildpipeline runs and removes it from file properties.  I'll look for a solution that is more dynamic but not on this commit.

File: AppInspector.CLI/AppInspector.CLI.csproj [hunk 5]
```diff
@@ -0,0 +1,153 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <None Remove="html\partials\_file_listing.liquid" />
```
[the remark this answers is not in the document]
Comment (guyacosta) [reply AppInspector.CLI/AppInspector.CLI.csproj]: system auto inserted both due to move from old to new project but verified the these section can be removed.  will fix in subsequent commit.

File: AppInspector/Commands/AllCommandOptions.cs [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+        [Option('i', "ignore-default-rules", Required = false, HelpText = "Exclude default rules bundled with application", Default = false)]
+        public bool IgnoreDefaultRules { get; set; }
+
+        [Option('o', "output-file-path", Required = false, HelpText = "Path to output file")]
+        public string OutputFilePath { get; set; }
+
+        [Option('x', "console-verbosity", Required = false, HelpText = "Console verbosity [high|medium|low|none]", Default = "medium")]
+        public string ConsoleVerbosityLevel { get; set; }
+
```
[the remark this answers is not in the document]
Comment (guyacosta) [reply AppInspector/Commands/AllCommandOptions.cs]: yep found one for VS2019.  thx.

File: RulesEngine/AppInspector.RulesEngine.csproj [hunk 7]
```diff
@@ -15,7 +15,7 @@
...
     <HighEntropyVA>true</HighEntropyVA>
-    <AssemblyName>RulesEngine</AssemblyName>
+    <AssemblyName>AppInspector.RulesEngine</AssemblyName>
     <RootNamespace>RulesEngine</RootNamespace>
```
[the remark this answers is not in the document]
Comment (guyacosta) [reply RulesEngine/AppInspector.RulesEngine.csproj]: Yes.  Focus was on restructure but updating the namespace would be an improvement.  Thx.  I'll address.

File: AppInspector/AppInspector.Lib.csproj [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    <PackageId>ApplicationInspector</PackageId>
+    <Product>Application Inspector</Product>
+    <Authors>Microsoft</Authors>
+    <Version>1.0.28</Version>
+    <AssemblyName>AppInspectorLib</AssemblyName>
+    <RootNamespace>ApplicationInspector</RootNamespace>
+    <StartupObject></StartupObject>
+    <Company>Microsoft</Company>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyVersion>1.0.28.0</AssemblyVersion>
```
[the remark this answers is not in the document]
Comment (guyacosta) [reply AppInspector/AppInspector.Lib.csproj]: Need it for now for reporting when running.  Will look for dynamic solution that addresses that aspect.

**Comment by guyacosta** [comment]
Going to merge this version with the embedded commit changes made in Github PR between Gabe and myself and resubmit a new PR with 1-2 additional improvements.  @daalcant you can review the next ver for pipeline updates needed based on new dll output file names.

**Pull request merged**